Source/inv.cpp

```diff
@@ -623,62 +623,11 @@ bool AutoEquipEnabled(const PlayerStruct &player, const ItemStruct &item)
  */
 bool AutoPlaceItemInInventory(PlayerStruct &player, const ItemStruct &item, bool persistItem)
 {
-	Size itemSize = GetInventorySize(item);
-
-	if (itemSize.Height == 1) {
-		for (int i = 30; i <= 39; i++) {
-			if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
-				return true;
-		}
-		for (int x = 9; x >= 0; x--) {
-			for (int y = 2; y >= 0; y--) {
-				if (AutoPlaceItemInInventorySlot(player, 10 * y + x, item, persistItem))
-					return true;
-			}
-		}
-		return false;
-	}
-
-	if (itemSize.Height == 2) {
-		for (int x = 10 - itemSize.Width; x >= 0; x -= itemSize.Width) {
-			for (int y = 0; y < 3; y++) {
-				if (AutoPlaceItemInInventorySlot(player, 10 * y + x, item, persistItem))
-					return true;
-			}
-		}
-		if (itemSize.Width == 2) {
-			for (int x = 7; x >= 0; x -= 2) {
-				for (int y = 0; y < 3; y++) {
-					if (AutoPlaceItemInInventorySlot(player, 10 * y + x, item, persistItem))
-						return true;
-				}
-			}
-		}
-		return false;
-	}
-
-	if (itemSize == Size { 1, 3 }) {
-		for (int i = 0; i < 20; i++) {
-			if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
-				return true;
-		}
-		return false;
-	}
-
-	if (itemSize == Size { 2, 3 }) {
-		for (int i = 0; i < 9; i++) {
-			if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
-				return true;
-		}
-
-		for (int i = 10; i < 19; i++) {
-			if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
-				return true;
-		}
-		return false;
+	for (int i = 0; i < NUM_INV_GRID_ELEM; i++) {
+		if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
+			return true;

```

Comment:
Doesn't this change the whole order where items are added? I think this was touched recently to enforce a given order.

#1876 

@Aneurin FYI

Reply:
so just reverse the loop

Reply:
> so just reverse the loop

I don't think it's as simple as that (it _might_ be... but I think there is more to it).

Reply:
Yeah this definitely looks like it would change the inventory filling algorithm. It's not just a question of whether to go forward or backwards, but also needs to consider the size of the item in order to group the right things together, plus gold and 1x1 items want to fill by columns once the bottom row is full.

Reply:
> Yeah this definitely looks like it would change the inventory filling algorithm. It's not just a question of whether to go forward or backwards, but also needs to consider the size of the item in order to group the right things together, plus gold and 1x1 items want to fill by columns once the bottom row is full.

there is 0 consistency in your algorithm, it skips valid slots, 2x2 items are being placed from the right side, everything else from the left, what's going on xD

Reply:
ree the pr and issue for further descriptions on how and why